xray/sql_go110.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2017-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with the License. A copy of the License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+
+// +build go1.10
+
+package xray
+
+import (
+	"context"
+	"database/sql/driver"
+	"sync"
+)
+
+// SQLConnector wraps the connector, and traces SQL executions.
+// Unlike SQLContext, SQLConnector doesn't filter the password of the dsn.
+// So, you have to filter the password before passing the dsn to SQLConnector.
+func SQLConnector(dsn string, connector driver.Connector) driver.Connector {
+	d := &driverDriver{
+		Driver:   connector.Driver(),
+		baseName: "unknown",
+	}
+	return &driverConnector{
+		Connector: connector,
+		driver:    d,
+		name:      dsn,
+		filtered:  true,
+		// initialized attr lazy because we have no context here.
+	}
+}
+
+func (conn *driverConn) ResetSession(ctx context.Context) error {
+	if sr, ok := conn.Conn.(driver.SessionResetter); ok {
+		return Capture(ctx, conn.attr.dbname, func(ctx context.Context) error {
+			conn.attr.populate(ctx, "RESET SESSION")
+			return sr.ResetSession(ctx)
+		})
+	}
+	return nil
+}
+
+type driverConnector struct {
+	driver.Connector
+	driver   *driverDriver
+	filtered bool
+	name     string
+
+	mu   sync.RWMutex
+	attr *dbAttribute
+}
+
+func (c *driverConnector) Connect(ctx context.Context) (driver.Conn, error) {
+	var rawConn driver.Conn
+	attr, err := c.getAttr(ctx)
+	if err != nil {
+		return nil, err
+	}
+	err = Capture(ctx, attr.dbname, func(ctx context.Context) error {
```

Comment:
We should name the subsegment properly otherwise all the subsegments will have the same name because we pass attr.dbname as the name of the subsegment.

Reply:
ok.
should I rename all subsegments(e.g. Ping, Exec, Query, etc.)?
currently, all subsegments have same name `attr.dbname`.

https://github.com/aws/aws-xray-sdk-go/blob/16d917bfd8d13242ae4f03807a10773a5e9cd468/xray/sql_go18.go#L33
https://github.com/aws/aws-xray-sdk-go/blob/16d917bfd8d13242ae4f03807a10773a5e9cd468/xray/sql_go18.go#L44

Reply:
Yes. That would be great!